test/extended/prometheus/collection_profiles.go

```diff
@@ -0,0 +1,257 @@
+
+		for _, profile := range collectionProfilesSupportedList {
+			err := r.makeCollectionProfileConfigurationFor(tctx, profile)
+			o.Expect(err).To(o.BeNil())
+
+			o.Eventually(func() error {
+				queryResponse, err := helper.RunQuery(tctx, r.pclient, fmt.Sprintf(vectorExpression, profile))
+				if err != nil {
+					return err
+				}
+				if len(queryResponse.Data.Result) == 0 {
+					return fmt.Errorf("no result found for profile %q", profile)
+				}
+
+				return nil
+			}).Should(o.BeNil())
+		}
+	})
+	g.It("should have at-least one implementation for each collection profile", func() {
+		for _, profile := range collectionProfilesSupportedList {
+			err := r.makeCollectionProfileConfigurationFor(tctx, profile)
+			o.Expect(err).To(o.BeNil())
+
+			o.Eventually(func() error {
+				monitors, err := r.fetchMonitorsFor([2]string{collectionProfileFeatureLabel, profile})
+				if err != nil {
+					return err
+				}
+				if len(monitors.Items) == 0 {
+					return fmt.Errorf("no monitors found with collection profile %q", profile)
+				}
+
+				return nil
+			}).Should(o.BeNil())
+		}
+	})
+	g.It("should hide a default-only metric when minimal collection profile is enabled", func() {
+		defaultOnlyMetric := "kube_deployment_status_replicas"
+
+		for i, profile := range []string{collectionProfileFull, collectionProfileMinimal} {
+			err := r.makeCollectionProfileConfigurationFor(tctx, profile)
+			o.Expect(err).To(o.BeNil())
+
+			o.Eventually(func() error {
+				defaultOnlyMetricQuery := fmt.Sprintf("absent(%s @ %d) == 1", defaultOnlyMetric, time.Now().Unix())
+				queryResponse, err := helper.RunQuery(tctx, r.pclient, defaultOnlyMetricQuery)
+				if err != nil {
+					return err
+				}
+				if i == 0 && len(queryResponse.Data.Result) != 0 {
+					return fmt.Errorf("expected %q to be present", defaultOnlyMetric)
+				}
+				if i == 1 && len(queryResponse.Data.Result) == 0 {
+					return fmt.Errorf("expected %q to be absent", defaultOnlyMetric)
+				}
+
+				return nil
+			}).Should(o.BeNil())
+		}
+	})
+	g.It("should revert back to default collection profile when none is specified", func() {
+		err := r.makeCollectionProfileConfigurationFor(tctx, collectionProfileNone)
+		o.Expect(err).To(o.BeNil())
+
+		for i, profile := range []string{collectionProfileMinimal, collectionProfileFull} {
+			o.Eventually(func() error {
+				monitors, err := r.fetchMonitorsFor([2]string{collectionProfileFeatureLabel, profile})
+				if err != nil {
+					return err
+				}
+				if i == 0 && len(monitors.Items) != 0 {
+					return fmt.Errorf("monitors found with collection profile %q", profile)
+				}
+				if i == 1 && len(monitors.Items) == 0 {
+					return fmt.Errorf("no monitors found with collection profile %q", profile)
+				}
+
+				return nil
+			}).Should(o.BeNil())
+
+		}
+	})
+})
+
+func (r *runner) fetchMonitorsFor(selectors ...[2]string) (*pov1api.ServiceMonitorList, error) {
```

Comment:
the function signature isn't really obvious. Also it doesn't seem that the first item varies (e.g. it's always `collectionProfileFeatureLabel`).